devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "ICub.h"
+
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+#include <yarp/os/BufferedPort.h>
+
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+using namespace wearable::devices;
+
+const std::string LogPrefix = "ICub :";
+
+class ICub::ICubImpl
+{
+public:
+    class ICubForceTorque6DSensor;
+
+    wearable::TimeStamp timeStamp;
+
+    size_t nSensors;
+    std::vector<std::string> sensorNames;
+
+    bool leftHandFTDataReceived  = false;
+    bool rightHandFTDataReceived = false;
+
+    template<typename T>
+    struct icubDeviceSensors
+    {
+        std::shared_ptr<T> icubSensor;
+        size_t index;
```

Comment:
It is not clear to me how `index` is used.

Reply:
At the moment `index` is not used. So removed that variable